HLTrigger/HLTfilters/plugins/HLT2L1TkMuonL1TkMuonMuRefDR.cc

```diff
@@ -103,49 +97,38 @@ bool HLT2L1TkMuonL1TkMuonMuRefDR::getCollections(edm::Event& iEvent,
     return false;
 }
 
-bool HLT2L1TkMuonL1TkMuonMuRefDR::computeDR(edm::Event& iEvent, l1t::TkMuonRef& r1, l1t::TkMuonRef& r2) const {
-  float muRef1_eta = 0.;
-  float muRef1_phi = 0.;
-  if (r1->muonDetector() != 3) {
-    if (r1->muRef().isNull())
-      return false;
-
-    muRef1_eta = r1->muRef()->hwEta() * 0.010875;
-    muRef1_phi = static_cast<float>(l1t::MicroGMTConfiguration::calcGlobalPhi(
-        r1->muRef()->hwPhi(), r1->muRef()->trackFinderType(), r1->muRef()->processor()));
-    muRef1_phi = muRef1_phi * 2. * M_PI / 576.;
-  } else {
-    if (r1->emtfTrk().isNull())
-      return false;
+std::pair<float, float> HLT2L1TkMuonL1TkMuonMuRefDR::convertEtaPhi(l1t::TkMuonRef& tkmu) const {
+  float muRefEta = 0.;
+  float muRefPhi = 0.;
 
-    muRef1_eta = r1->emtfTrk()->Eta();
-    muRef1_phi = angle_units::operators::convertDegToRad(r1->emtfTrk()->Phi_glob());
-  }
-  muRef1_phi = reco::reduceRange(muRef1_phi);
-
-  float muRef2_eta = 0.;
-  float muRef2_phi = 0.;
-  if (r2->muonDetector() != 3) {
-    if (r2->muRef().isNull())
-      return false;
-
-    muRef2_eta = r2->muRef()->hwEta() * 0.010875;
-    muRef2_phi = static_cast<float>(l1t::MicroGMTConfiguration::calcGlobalPhi(
-        r2->muRef()->hwPhi(), r2->muRef()->trackFinderType(), r2->muRef()->processor()));
-    muRef2_phi = muRef2_phi * 2. * M_PI / 576.;
+  if (tkmu->muonDetector() != emtfRegion_) {
+    if (tkmu->muRef().isNull())
+      return std::make_pair(muRefEta, muRefPhi);
+
+    muRefEta = tkmu->muRef()->hwEta() * etaScale_;
+    muRefPhi = static_cast<float>(l1t::MicroGMTConfiguration::calcGlobalPhi(
+        tkmu->muRef()->hwPhi(), tkmu->muRef()->trackFinderType(), tkmu->muRef()->processor()));
+    muRefPhi = muRefPhi * phiScale_;
   } else {
-    if (r2->emtfTrk().isNull())
-      return false;
+    if (tkmu->emtfTrk().isNull())
+      return std::make_pair(muRefEta, muRefPhi);
 
-    muRef2_eta = r2->emtfTrk()->Eta();
-    muRef2_phi = angle_units::operators::convertDegToRad(r2->emtfTrk()->Phi_glob());
+    muRefEta = tkmu->emtfTrk()->Eta();
+    muRefPhi = angle_units::operators::convertDegToRad(tkmu->emtfTrk()->Phi_glob());
   }
-  muRef2_phi = reco::reduceRange(muRef2_phi);
+  muRefPhi = reco::reduceRange(muRefPhi);
+
+  return std::make_pair(muRefEta, muRefPhi);
+}
 
-  if (reco::deltaR(muRef1_eta, muRef1_phi, muRef2_eta, muRef2_phi) > minDR_)
+bool HLT2L1TkMuonL1TkMuonMuRefDR::computeDR(edm::Event& iEvent, l1t::TkMuonRef& r1, l1t::TkMuonRef& r2) const {
+  if (minDR_ < 0.)
     return true;
 
-  return false;
+  std::pair<float, float> muRef1 = convertEtaPhi(r1);
+  std::pair<float, float> muRef2 = convertEtaPhi(r2);
+
+  return (reco::deltaR2(muRef1.first, muRef1.second, muRef2.first, muRef2.second) > minDR_ * minDR_);
```

Comment:
a suggestion, if you think it would be more clear:
```suggestion
  auto [eta1, phi1] = convertEtaPhi(r1);
  auto [eta2, phi2] = convertEtaPhi(r2);
  return (reco::deltaR2(eta1, phi1, eta2, phi2) > minDR_ * minDR_);
```